membership_initial_fee/models/account_invoice.py

```diff
@@ -32,18 +32,28 @@ def _prepare_initial_fee_vals(self, invoice_line):
         return line_vals
 
     @api.model
+    def initial_fee_add(self, product, invoice_line):
+        """Inherit this method to implement a custom method
+           to decide whether or not to create the initial fee
+        """
+        if not product.membership or product.initial_fee == 'none':
+            return False
+        # See if partner has any membership line to decide whether or not
+        # to create the initial fee
+        member_lines = self.env['membership.membership_line'].search([
+            ('partner', '=', invoice_line.invoice_id.partner_id.id),
+            ('account_invoice_line', 'not in', (invoice_line.id, )),
+            ('state', 'not in', ['none', 'canceled']),
+        ])
+        if not member_lines:
+            # Charge initial fee
+            return self.create(self._prepare_initial_fee_vals(invoice_line))
+        return False
+
+    @api.model
     def create(self, vals):
         invoice_line = super(AccountInvoiceLine, self).create(vals)
         product = self.env['product.product'].browse(
             vals.get('product_id'))
```

Comment:
product_id is not required on invoice line. to avoid a bug on line without product_id, It will be nice to test if product_id exist before pass to the rest of code.